camayoc/tests/qpc/cli/test_openshift.py

```diff
@@ -141,6 +141,9 @@ def test_openshift_clusters(qpc_server_config, data_provider, source_definition:
     :expectedresults: The facts already knew about the cluster and nodes
         have expected values in deployment and details reports.
     """
+    cred = data_provider.credentials.defined_one({"name": source_definition.name})
```

Comment:
But that's going to work only by a chance, right? When it happens that credential has the same name as a source.

I feel this logic should be moved to `openshift_sources()` - go through sources, for each source check all credentials, and check if there's at least one credential that has either token or password. If there is none, then source is skipped.

If there is no sources matching these criteria, then `openshift_sources()` will return empty iterable, effectively skipping the test. The only drawback is that this is not very explicit - the test just does not exist instead of being skipped.

I don't know if you can skip test from parameter iterable. Maybe you can, in which case we could call out skip explicitly and get the best of both worlds.